src/js.c

```diff
@@ -842,19 +842,23 @@ _timeoutPubAsync(natsTimer *t, void *closure)
 
     while (((pm = js->pmHead) != NULL) && (pm->deadline <= now))
     {
-        natsMsg *m = NULL;
-
-        if (natsMsg_Create(&m, pm->subject, NULL, NULL, 0) != NATS_OK)
-            break;
-
-        natsMsg_setTimeout(m);
+        // Check if the corresponding message is still in the hashtable.
+        char *id = (pm->subject+js->rpreLen);
+        if (natsStrHash_Get(js->pm, id) != NULL)
```

Comment:
Should this be == NULL?

Reply:
Or does this mean only create the new message if we have an antry, otherwise do not create one at all?